esphome/components/hitachi_ac344/hitachi_ac344.cpp

```diff
@@ -0,0 +1,395 @@
+        this->mode = climate::CLIMATE_MODE_AUTO;
+        break;
+      case HITACHI_AC344_MODE_FAN:
+        this->mode = climate::CLIMATE_MODE_FAN_ONLY;
+        break;
+    }
+  } else {
+    this->mode = climate::CLIMATE_MODE_OFF;
+  }
+  return true;
+}
+
+bool HitachiClimate::parse_temperature_(const uint8_t remote_state[]) {
+  uint8_t temperature =
+      GETBITS8(remote_state[HITACHI_AC344_TEMP_BYTE], HITACHI_AC344_TEMP_OFFSET, HITACHI_AC344_TEMP_SIZE);
+  this->target_temperature = temperature;
+  ESP_LOGV(TAG, "Temperature: %02X %02u %04f", remote_state[HITACHI_AC344_TEMP_BYTE], temperature,
+           this->target_temperature);
+  return true;
+}
+
+bool HitachiClimate::parse_fan_(const uint8_t remote_state[]) {
+  uint8_t fan_mode = remote_state[HITACHI_AC344_FAN_BYTE] >> 4 & 0xF;
+  ESP_LOGV(TAG, "Fan: %02X %02X", remote_state[HITACHI_AC344_FAN_BYTE], fan_mode);
+  switch (fan_mode) {
+    case HITACHI_AC344_FAN_MIN:
+    case HITACHI_AC344_FAN_LOW:
+      this->fan_mode = climate::CLIMATE_FAN_LOW;
+      break;
+    case HITACHI_AC344_FAN_MEDIUM:
+      this->fan_mode = climate::CLIMATE_FAN_MEDIUM;
+      break;
+    case HITACHI_AC344_FAN_HIGH:
+    case HITACHI_AC344_FAN_MAX:
+      this->fan_mode = climate::CLIMATE_FAN_HIGH;
+      break;
+    case HITACHI_AC344_FAN_AUTO:
+      this->fan_mode = climate::CLIMATE_FAN_AUTO;
+      break;
+  }
+  return true;
+}
+
+bool HitachiClimate::parse_swing_(const uint8_t remote_state[]) {
+  // uint8_t swing_modev = GETBIT8(remote_state[HITACHI_AC344_SWINGV_BYTE],
+  //                HITACHI_AC344_SWINGV_OFFSET);
+  // ESP_LOGV(TAG, "SwingV: %02X %02X",
+  //          remote_state[HITACHI_AC344_SWINGV_BYTE], swing_modev);
+  uint8_t swing_modeh =
+      GETBITS8(remote_state[HITACHI_AC344_SWINGH_BYTE], HITACHI_AC344_SWINGH_OFFSET, HITACHI_AC344_SWINGH_SIZE);
+  ESP_LOGV(TAG, "SwingH: %02X %02X", remote_state[HITACHI_AC344_SWINGH_BYTE], swing_modeh);
+
+  if ((swing_modeh & 0x7) == 0x0) {
+    this->swing_mode = climate::CLIMATE_SWING_HORIZONTAL;
+  } else if ((swing_modeh & 0x3) == 0x3) {
+    this->swing_mode = climate::CLIMATE_SWING_OFF;
+  } else {
+    this->swing_mode = climate::CLIMATE_SWING_HORIZONTAL;
+  }
+  // if (swing_modev & 0x20 && swing_modeh & 0x7 == 0x0)
+  //   this->swing_mode = climate::CLIMATE_SWING_BOTH;
+  // else if (swing_modev & 0x20)
+  //   this->swing_mode = climate::CLIMATE_SWING_VERTICAL;
+  // else if (swing_modeh & 0x3 == 0x3)
+  //   this->swing_mode = climate::CLIMATE_SWING_HORIZONTAL;
+  // else
+  //   this->swing_mode = climate::CLIMATE_SWING_OFF;
+  // remote_state[HITACHI_AC344_BUTTON_BYTE] = HITACHI_AC344_BUTTON_SWINGV
+  // remote_state[HITACHI_AC344_BUTTON_BYTE] = HITACHI_AC344_BUTTON_POWER
+
+  return true;
+}
+
+bool HitachiClimate::on_receive(remote_base::RemoteReceiveData data) {
+  // Validate header
+  if (!data.expect_item(HITACHI_AC344_HDR_MARK, HITACHI_AC344_HDR_SPACE)) {
+    ESP_LOGV(TAG, "Header fail");
```

Comment:
```suggestion
    ESP_LOGVV(TAG, "Header fail");
```

Change those to very verbose, this will be logged even when stuff not related to this integration is received.